### ISSUE TITLE
How to view train error metrics for a model

### ISSUE THREAD

**Issue by mattdeeperinsights** [aws/sagemaker-huggingface-inference-toolkit#47]
Thanks for the package, I have enjoyed using it greatly but have a question about metrics.

I have trained a model [using the Hello World example for token classification](https://huggingface.co/docs/sagemaker/train).

I can easily calculate and view the metrics generated on the evaluation test set: accuracy, f-score, precision, recall etc. by calling `training_job_analytics` on the trained model: `huggingface_estimator.training_job_analytics.dataframe()`

How can I also see the same metrics on training sets (or even training error for each epoch)?

Training code is basically the same as the link with extra parts of the docs added:

```python3
from sagemaker.huggingface import HuggingFace

# optionally parse logs for key metrics
# from the docs: https://huggingface.co/docs/sagemaker/train#sagemaker-metrics
metric_definitions = [
    {'Name': 'loss', 'Regex': "'loss': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'learning_rate', 'Regex': "'learning_rate': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_loss', 'Regex': "'eval_loss': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_accuracy', 'Regex': "'eval_accuracy': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_f1', 'Regex': "'eval_f1': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_precision', 'Regex': "'eval_precision': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_recall', 'Regex': "'eval_recall': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_runtime', 'Regex': "'eval_runtime': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'eval_samples_per_second', 'Regex': "'eval_samples_per_second': ([0-9]+(.|e\-)[0-9]+),?"},
    {'Name': 'epoch', 'Regex': "'epoch': ([0-9]+(.|e\-)[0-9]+),?"}
]

# hyperparameters, which are passed into the training job
hyperparameters={
    'epochs': 5,
    'train_batch_size': batch_size,
    'model_name': model_checkpoint,
    'task': task,
}

# init the model (but not yet trained)
huggingface_estimator = HuggingFace(
    entry_point='train.py',
    source_dir='./scripts',
    instance_type='ml.p3.2xlarge',
    instance_count=1,
    role=role,
    transformers_version='4.6',
    pytorch_version='1.7',
    py_version='py36',
    hyperparameters = hyperparameters,
    metric_definitions=metric_definitions
)
# starting the train job with our uploaded datasets as input
huggingface_estimator.fit({'train': training_input_path, 'test': test_input_path})

# does not return metrics on training - only on eval!
huggingface_estimator.training_job_analytics.dataframe()
```

This is my output:

![image](https://user-images.githubusercontent.com/90614572/145981332-083e28f3-45e1-4e7e-8a0b-52e89f29b580.png)

as a json:

```python3
[{'timestamp': 0.0, 'metric_name': 'eval_loss', 'value': 2.09915896824428},
 {'timestamp': 0.0,
  'metric_name': 'eval_accuracy',
  'value': 0.48287172011661805},
 {'timestamp': 0.0, 'metric_name': 'eval_f1', 'value': 0.0023274664199115004},
 {'timestamp': 0.0,
  'metric_name': 'eval_precision',
  'value': 0.0012611563074355545},
 {'timestamp': 0.0,
  'metric_name': 'eval_recall',
  'value': 0.02029664324746292},
 {'timestamp': 0.0,
  'metric_name': 'eval_runtime',
  'value': 0.19744285714285711},
 {'timestamp': 0.0,
  'metric_name': 'eval_samples_per_second',
  'value': 51.15857142857143},
 {'timestamp': 0.0, 'metric_name': 'epoch', 'value': 11.857142857142858}]
```

**Comment by mattdeeperinsights** [comment]
I think I have solved this by increasing the number of epochs to 1000 from 20, now I get all of the things that I needed
